app/code/Magento/Store/etc/config.xml

```diff
@@ -84,6 +84,7 @@
                 <use_http_x_forwarded_for>0</use_http_x_forwarded_for>
                 <use_http_user_agent>0</use_http_user_agent>
                 <use_frontend_sid>1</use_frontend_sid>
+                <old_session_access_delta>300</old_session_access_delta>
```

Comment:
I don't think putting such a sensitive configuration option into Admin UI is a good idea. 300 seconds is a huge delay, I believe we should pick up some reasonable value.

Ideally solution must not involve any timeout at all allowing just to create new session seamlessly.

Reply:
@orlangur, thank you for the review.
I agree that such configuration should not be placed in Admin UI.

About timeout, I have added it to the solution as additional security frontier in session recovery.
Old session will be removed as soon as response with new session id will be received.
And this timeout is for that cases when new session was generated but response with new session id will never be received.

Unfortunately, one of PHP session flaw is that you can not be sure that cookie with session id was delivered to browser or no.

About timeout, I agree that 5m is a huge delay.  I have decreased the default number to 30s.